File: Data Structures & Algorithms/missing-number/submission-2.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        long long expectedsum = (long long)n*(n+1)/2;
        long long actualsum = 0;

        for(int num:nums){
            actualsum += num;
        }
        return expectedsum - actualsum;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        int n = nums.length;
        Arrays.sort(nums);

        HashMap<Integer,Integer>mp = new HashMap<>();
        for(int num:nums){
            mp.put(num,mp.getOrDefault(num,0)+1);
        }

        List<Integer>list = new ArrayList<>(mp.keySet());
        list.sort((a,b) -> mp.get(b) - mp.get(a));

        int[] result = new int[k];

        for(int i=0;i<k;i++){
            result[i] = list.get(i);
        }
        return result;
    }
}
